[12.0][ADD] timesheet_helpdesk

oscars8a opened the pull request:
Add timesheet for Helpdesk Management (helpdesk_mgmt).

SPANISH OCA Days 2020

EDallas89 committed:
[ADD] timesheet_helpdesk

DarioLodeiros commented on timesheet_helpdesk_ticket/README.md:
```diff
@@ -0,0 +1,6 @@
+                           Simple Timesheet for Helpdesk's Tickets
```
README with guideline OCA

oscars8a replied:
Readme fixed with OCA guidelines

DarioLodeiros commented on timesheet_helpdesk_ticket/__manifest__.py:
```diff
@@ -0,0 +1,19 @@
+{
+    'name': "Timesheet for Helpdesk's Tickets",
+    'summary': """
+        Añade las funciones de partes de horas a los tickets de Helpdesk Management """,
```
Review guidelines and not use Spanish lang

oscars8a replied:
Fixed.
'summary': 'Add HR Timesheet to the tickets for Helpdesk Management.',

DarioLodeiros commented on timesheet_helpdesk_ticket/__manifest__.py:
```diff
@@ -0,0 +1,19 @@
+    'summary': """
+        Añade las funciones de partes de horas a los tickets de Helpdesk Management """,
+    'author': "Inma Sánchez y Carlos Ramos",
+    'website': "https://github.com/EDallas89",
```
OCA Guidelines

oscars8a replied:
'website': 'https://github.com/oca/helpdesk',

DarioLodeiros commented on timesheet_helpdesk_ticket/__manifest__.py:
```diff
@@ -0,0 +1,19 @@
+    'version': '12.0',
+    'depends': ['helpdesk_mgmt', 'hr_timesheet','resource',],
+    'data': [
+        #'security/ir.model.access.csv',
```
DEL line

oscars8a replied:
Removed it.

DarioLodeiros commented on timesheet_helpdesk_ticket/models/helpdesk_ticket_team.py:
```diff
@@ -0,0 +1,26 @@
+class HelpdeskTicketTeam(models.Model):
+    _inherit = 'helpdesk.ticket.team'
+
+    archive = fields.Boolean('Archive', related='active', default=True, store=True, readonly=False)    
```
This is out-topic

oscars8a replied:
Removed for this PR, this change is in https://github.com/OCA/helpdesk/pull/126

DarioLodeiros commented on timesheet_helpdesk_ticket/models/helpdesk_ticket_team.py:
```diff
@@ -0,0 +1,26 @@
+    allow_timesheet = fields.Boolean(
+        string="Allow Timesheet",
+    )
+    default_project = fields.Many2one(
```
Separate project logic from the helpdesk timesheet logic

EDallas89 replied:
Changed to Analytic Account

DarioLodeiros commented on timesheet_helpdesk_ticket/models/ticket_timesheet.py:
```diff
@@ -0,0 +1,59 @@
+class TicketTimesheet(models.Model):
+    _inherit = 'helpdesk.ticket'
+
+    partner_phone = fields.Char(
```
Off topic on this PR

EDallas89 replied:
Removed on  #121

DarioLodeiros commented on timesheet_helpdesk_ticket/models/ticket_timesheet.py:
```diff
@@ -0,0 +1,59 @@
+        store=True,
+        readonly=False,
+    )
+    project_id = fields.Many2one(
```
No project logic on timesheets helpdesk

DarioLodeiros commented on timesheet_helpdesk_ticket/models/ticket_timesheet.py:
```diff
@@ -0,0 +1,59 @@
+        inverse_name='ticket_id',
+        string='Timesheet',
+    )
+    total_hours = fields.Float(
```
This logic already exist on account.analytic.line

oscars8a replied:
Total hours is on the ticket not on the analytic.line...

DarioLodeiros commented on timesheet_helpdesk_ticket/models/ticket_timesheet.py:
```diff
@@ -0,0 +1,59 @@
+        comodel_name='helpdesk.ticket.team',
+    )
+
+    @api.depends('timesheet_ids.unit_amount')
```
I don't see why we need this here (view total_hours field comment)

oscars8a replied:
Fixed it.

DarioLodeiros commented on timesheet_helpdesk_ticket/views/helpdesk_team.xml:
```diff
@@ -0,0 +1,26 @@
+        <field name="priority" eval="20" />
+        <field name="arch" type="xml">
+            <!-- Only For Fix Smart Button-->
+            <xpath expr="//div[@name='button_box']" position="replace">
```
Off topic.
This has to be a PR on heldesk_mgmt

oscars8a replied:
Fixed it.

DarioLodeiros commented on timesheet_helpdesk_ticket/views/helpdesk_ticket.xml:
```diff
@@ -0,0 +1,77 @@
+                    </h2>
+                </div>
+            </xpath>
+            <xpath expr="//group[@name='main']" position="replace">
```
Remove (off topic)

oscars8a replied:
Fixed it.

Commit:
[FIX] module name and manifest

Commit:
[FIX] remove resource dependencie

oscars8a committed:
Merge pull request #5 from oscars8a/12.0-spanish_oca_days

[FIX] module name and manifest

oscars8a committed:
Merge pull request #7 from oscars8a/12.0-spanish_oca_days

12.0 spanish oca days

EDallas89 committed:
[REM] Removing Extra Code

Removing changes on smart button (helpdesk.team)
Removing changes on order fields (helpesk.ticket)

EDallas89 committed:
Merge branch '12.0.helpdesk.timesheet' of https://github.com/oscars8a/helpdesk-1 into 12.0.helpdesk.timesheet

pedrobaeza commented:
Some comments in a quick sight:

- Module name should be `helpdesk_mgmt_timesheet`, for following module base name (and avoid overlapping with a future Odoo module), and for being more general (although right now the timesheeting is only done on tickets, it may be in other places/screens).
- README should generated from fragments and in RST format.

pedrobaeza commented:
Check also Travis state

EDallas89 committed:
[REF] Change Project Dependencies to Analytic Account

EDallas89 committed:
Update Authors

EDallas89 committed:
Update __manifest__.py

EDallas89 committed:
[IMP] Update Readme

EDallas89 committed:
Merge branch '12.0.helpdesk.timesheet' of https://github.com/oscars8a/helpdesk-1 into 12.0.helpdesk.timesheet

JorgeQuinteros committed:
Merge branch '12.0' of https://github.com/oscars8a/helpdesk-1 into 12.0.helpdesk.timesheet

JorgeQuinteros committed:
Merge branch '12.0.helpdesk.timesheet' of https://github.com/oscars8a/helpdesk-1 into 12.0.helpdesk.timesheet

JorgeQuinteros committed:
[FIX] Cleancode Flake-8

JorgeQuinteros committed:
Merge pull request #10 from oscars8a/12.0-cleancode

12.0 cleancode

EDallas89 committed:
[IMP] Little Changes

Change to invisible account_id field on timesheet.
Add help on default_analytic_account field on teams.

Commit:
[ADD]Copyright Punt Sistemes

Commit:
[ADD]Copyright Punt Sistemes

Commit:
[FIX] helpdesk ticekt with timesheet

Commit:
[FIX] helpdesk ticekt with timesheet

Commit:
Merge branch '12.0-spanish_oca_days' of https://github.com/oscars8a/helpdesk-1 into 12.0-spanish_oca_days

Commit:
[FIX] flake8

JorgeQuinteros committed:
Merge pull request #11 from EDallas89/12.0.helpdesk.timesheet

[IMP] Little Changes

JorgeQuinteros committed:
Merge pull request #12 from oscars8a/12.0-spanish_oca_days

12.0 spanish oca days

JorgeQuinteros committed:
[FIX] module name

emagdalenaC2i commented:
Please, use the fragmented README. See https://github.com/OCA/maintainer-tools/tree/master/template/module

JorgeQuinteros committed:
Merge pull request #13 from oscars8a/12.0-helpdesk_mgmt_timesheet

[FIX] module name

JorgeQuinteros committed:
[i18n] Improvement translations

oscars8a committed:
Merge pull request #15 from oscars8a/12.0-translations

[i18n] Improvement translations

Commit:
[DEL] csv

EDallas89 committed:
[IMP] Readme

Commit:
[ADD] test

JorgeQuinteros committed:
Merge pull request #16 from EDallas89/12.0-helpdesk_mgmt_timesheet

[IMP] Readme

EDallas89 committed:
[IMP] Invisible Timesheet

Add a condition to show timesheet table.
Timesheet is invisible if analytic_account_id field is empty.

JorgeQuinteros committed:
[FIX] hide analytic account if the team doesn't have the option to Allow Timesheet

JorgeQuinteros committed:
Merge pull request #18 from EDallas89/12.0-helpdesk_mgmt_timesheet

[IMP] Invisible Timesheet

JorgeQuinteros committed:
Merge pull request #19 from oscars8a/12.0-helpdesk_ticket_view

[ADD] hide analytic account if the team doesn't have the option to Al…

Commit:
[FIX] pep8

JorgeQuinteros committed:
Merge pull request #17 from oscars8a/12.0-spanish_oca_days

[ADD]test and remove csv

emagdalenaC2i commented:
Please, add more info in the USAGE file about allow Timesheet
![image](https://user-images.githubusercontent.com/24428859/82362104-1cb30200-9a0c-11ea-8343-b73de916e30d.png)

EDallas89 commented:
> Please, add more info in the USAGE file about allow Timesheet
> ![image](https://user-images.githubusercontent.com/24428859/82362104-1cb30200-9a0c-11ea-8343-b73de916e30d.png)
Info about Allow Timesheet field is on CONFIGURE file.

JorgeQuinteros committed:
[ADD] complete name of contributor SDi

EDallas89 committed:
[IMP] Total Hours Stored

EDallas89 committed:
[IMP] Complete Name of Autor Aresoltec

JorgeQuinteros reviewed (approved):
Technical review

JGarcia-Panach reviewed (approved):
👍  Functional review

oscars8a committed:
Merge pull request #20 from EDallas89/12.0-helpdesk_mgmt_timesheet

[IMP] Total Hours Stored

JorgeQuinteros reviewed (approved):
Technical review.

Commit:
[FIX] tests

Commit:
[FIX] test timesheet

marcelsavegnago reviewed (approved):
👍 Functional review

pedrobaeza commented on helpdesk_mgmt_timesheet/__manifest__.py:
```diff
@@ -0,0 +1,39 @@
+#    Copyright (C) 2020 Punt Sistemes <www.puntsistemes.es.es>
+#    Copyright (C) 2020 SDi Soluciones Digitales <www.sdi.es>
+#
+#    This program is free software: you can redistribute it and/or modify
```
These bloated headers are obsolete since a lot. Please use the reduced ones:

https://github.com/OCA/maintainer-tools/blob/master/template/module/models/model_name.py#L1-L2

oscars8a replied:
Fixed!

pedrobaeza commented on helpdesk_mgmt_timesheet/readme/CONFIGURE.rst:
```diff
@@ -0,0 +1,11 @@
+To configure this module, you need to:
+#. Allow Timesheet for a Helpdesk's Team
```
Need empty line between for proper RST syntax

oscars8a replied:
Fixed, thanks a lot.

DarioLodeiros reviewed (changes requested):
This module should depend on #129 
The relationship between ticket and analytic account is given by project, following the guidelines given by the Core (project is a "required" -not in model, but yes on views- field on analytic account line). If we use the project to link this two models, we will have access to all the functions developed on this

Commit:
[FIX] bloated headers

Commit:
[FIX] empty line between for proper RST syntax

Commit:
[FIX] add empty end line

dalonsod reviewed (changes requested):
I miss `ticket_id` references in general `account.analytic.line` tree view and report. Could they be included?

dalonsod commented:
> 
> 
> This module should depend on #129
> The relationship between ticket and analytic account is given by project, following the guidelines given by the Core (project is a "required" -not in model, but yes on views- field on analytic account line). If we use the project to link this two models, we will have access to all the functions developed on this

@oscars8a #129 is already merged, it's time to resume this PR! If help is needed, I could contribute 😄

dalonsod commented:
@oscars8a any news on this?

dalonsod commented:
@DarioLodeiros this PR seems on an idle state. If there's no problem, I could resume it and supersede the PR

DarioLodeiros commented:
Perfect @dalonsod

dalonsod commented:
@DarioLodeiros as #151 is already finished, this PR should be closed

DarioLodeiros commented:
sure, thanks!
